Frends.SFTP.ListFiles/Frends.SFTP.ListFiles/Frends.SFTP.ListFiles.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.ComponentModel;
+using Renci.SshNet;
+using System.Net.Sockets;
+using Renci.SshNet.Common;
+using System.Text.RegularExpressions;
+using Frends.SFTP.ListFiles.Definitions;
+
+#pragma warning disable 1591
+
+namespace Frends.SFTP.ListFiles
+{
+    public class SFTP
+    {
+        /// <summary>
+        /// Reads a file with SFTP connection.
+        /// Documentation: https://github.com/FrendsPlatform/Frends.SFTP.ReadFile
+        /// </summary>
+        /// <param name="connection">Transfer connection parameters</param>
+        /// <param name="options">Source file location</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>List [ Object { string FullPath, bool IsDirectory, bool IsFile, long Length, string Name, DateTime LastWriteTimeUtc, DateTime LastAccessTimeUtc, DateTime LastWriteTime, DateTime LastAccessTime } ]</returns>
+        public static List<Result> ListFiles([PropertyTab] Options options, [PropertyTab] Connection connection, CancellationToken cancellationToken)
+        {
+            // Establish connectionInfo with connection parameters
+            var connectionInfo = GetConnectionInfo(connection);
+            var result = new List<Result>();
+            var regexStr = string.IsNullOrEmpty(options.FileMask) ? string.Empty : WildCardToRegex(options.FileMask);
+
+            try
+            {
+                using (var client = new SftpClient(connectionInfo))
+                {
+                    client.Connect();
+                    result = GetFiles(client, regexStr, options.Directory, options, cancellationToken);
+                    client.Disconnect();
+
+                }
+            }
+            catch (SshConnectionException ex)
+            {
+                throw new Exception($"Error when establishing connection to the Server: {ex.Message}", ex);
+            }
+            catch (SocketException ex)
+            {
+                throw new Exception($"Unable to establish the socket: No such host is known.", ex);
+            }
+            catch (SshAuthenticationException ex)
+            {
+                throw new Exception($"Authentication of SSH session failed: {ex.Message}", ex);
+            }
+
+            return result;
+        }
+
+        private static List<Result> GetFiles(SftpClient sftp, string regexStr, string directory, Options options, CancellationToken cancellationToken)
+        {
+            var directoryList = new List<Result>();
+
+            var files = sftp.ListDirectory(directory);
+
+            foreach (var file in files)
+            {
+                if (file.Name != "." && file.Name != "..")
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This token check should be moved above the if-statement, since if the execution of the iteration doesn't go here, the token will not be checked and it should be check on each iteration.

Reply:
Fixed